src/elektra/redshift.ini

```diff
@@ -0,0 +1,233 @@
+opt/arg = required
+
+[gamma/day]
+type = string
+description = The screen gamma during daytime.
+default = 1.0:1.0:1.0
+example = 0.9:0.9:0.9
+; Regex checks that value is of format "float:float:float". Adapted from https://www.regular-expressions.info/floatingpoint.html
+check/validation = ^([0-9]*[\.,]?[0-9]+):([0-9]*[\.,]?[0-9]+):([0-9]*[\.,]?[0-9]+)$
+opt/long = gamma-day
+opt/arg = required
+
+[gamma/night]
+type = string
+description = The screen gamma during nighttime.
+default = 1.0:1.0:1.0
+example = 0.9:0.9:0.9
+; Regex checks that value is of format "float:float:float". Adapted from https://www.regular-expressions.info/floatingpoint.html
+check/validation = ^([0-9]*[\.,]?[0-9]+):([0-9]*[\.,]?[0-9]+):([0-9]*[\.,]?[0-9]+)$
+opt/long = gamma-night
+opt/arg = required
+
+[adjustment/method]
+type = string
+check/enum = #5
+check/enum/#0 = drm
+check/enum/#1 = dummy
+check/enum/#2 = quartz
+check/enum/#3 = randr
+check/enum/#4 = vidmode
+check/enum/#5 = w32gdi
+description = The method used to adjust screen color temperature. For details see section "Alternative Features" in file DESIGN in root directory.
+require =
+example = randr
+; default = w32gdi - For Windows the default is the only supported method 'w32gdi'
+; default = randr - For Linux the default is 'randr'
+; default = quartz - For macOS the default is 'quartz'
+opt = m
+opt/long = method
+opt/arg = required
+
+[provider]
+type = string
+check/enum = #1
+check/enum/#0 = time
+check/enum/#1 = location
+description = The provider used to decide at what times of day redshift should be enabled/disabled. Currently two options are supported: 1. location - determines the user's location and enable/disable redshift depending on the solar elevation. 2. time: Ignore user location and enable redshift if time is between dusk and dawn.
+default = location
+example = time
+opt/long = provider
+opt/arg = required
+
+[provider/time/dawn]
+type = string
+check/date = ISO8601
+check/date/format = timeofday
+description = Instead of using the solar elevation at the user's location, the time intervals of dawn and dusk can be specified manually (source: https://github.com/jonls/redshift/wiki/Configuration-file#custom-dawndusk-intervals).
+default = 05:00
+example = 06:00
+opt/long = time-dawn
+opt/arg = required
+
+[provider/time/dusk]
+type = string
+check/date = ISO8601
+check/date/format = timeofday
+description = Instead of using the solar elevation at the user's location, the time intervals of dawn and dusk can be specified manually (source: https://github.com/jonls/redshift/wiki/Configuration-file#custom-dawndusk-intervals).
+default = 19:00
+example = 20:00
+opt/long = time-dusk
+opt/arg = required
+
+[provider/location]
+type = string
+check/enum = #2
+check/enum/#0 = corelocation
+check/enum/#1 = geoclue2
+check/enum/#2 = manual
+description = The location provider to be used. The provider is used to establish whether it is currently daytime or nighttime.
+default = manual
```

Comment:
Similar to my comment on "adjustment/method", this one should perhaps default to trying any available location provider? This one is simpler because there's only one possible method for each platform `corelocation` = MacOS, `geoclue2` = Linux, Windows doesn't have a location provider, but it's not necessarily compiled in.

Reply:
Great idea, I've incorporated that.